cs/src/core/protocols/CompactBinary.cs

```diff
@@ -176,7 +176,11 @@ public IProtocolWriter GetFirstPassWriter()
         {
             if (version == 2)
             {
-                return firstPassWriter.Value;
+                // Only return first pass if not in middle of second pass
+                if (lengths.Count == 0)
```

Comment:
I checked to confirm. `Count` on a `LinkedList` is [constant time](https://msdn.microsoft.com/en-us/library/t56y0h99(v=vs.110).aspx#Anchor_1).